keps/sig-api-machinery/4358-custom-resource-field-selectors/README.md

```diff
@@ -0,0 +1,1150 @@
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation_test.go
+  - [x] duplicate selectableFields not allowed
+  - [x] invalid path not allowed
+  - [x] path to invalid type not allowed
+  - [x] too many selectableFields not allowed
+  - [x] valueIfAbsent required if field may be absent
+  - [x] valueIfAbsent not allowed if field cannot be absent
+
+- staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy_test.go
+  - [ ] ensure field paths are not dereferenced if not used, or make it fast and benchmark it
+  - [ ] all possible type cases, including string format cases with special type bindings (duration, intOrString)
+
+- staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/strategy_test.go
+  - [x] all drop field cases
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- test/integration/controlplane/crd_test.go
+  - TODO: Test selectable field {with, without valueIsAbsent}, {with single field, with multiple fields, with combined use of label selectors} for {list, watch}
```

Comment:
also test when a watched item changes values for its selection.  Does a delete get issued to the watch?